android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -189,6 +197,11 @@ private Byte getType(Object value) {
       return PlatformConstants.CodecTypes.pushChannelSubscription;
     } else if (value instanceof RemoteMessage) {
       return PlatformConstants.CodecTypes.remoteMessage;
+    } else if (value instanceof Crypto.CipherParams) {
+      return PlatformConstants.CodecTypes.cipherParams;
+    } else if (value instanceof ChannelOptions) {
```

Comment:
Where did you find `RealtimeChannelOptions` in Ably Java? I could not find such symbol, and concluded `ChannelOptions` is shared between Realtime and Rest channels.

However, because you mentioned it, I had a deeper read into Ably Java and found a different issue: https://github.com/ably/ably-java/issues/730

Reply:
I have not found `RealtimeChannelOptions` anywhere. As I'm doing light review only, I'm skimming through your code. Since you are concluding that the result should translate into `realtimeChannelOptions` but only evaluating a base class might be risky. So in this case, if you are sure thhat this will only translate to realtimeChannelOptions and not `restChannelOptions`(I don't know if this exist as well) it's fine.

Reply:
All this code was removed in 8ecfce0cfe1eeb3c511ca8b4b6f039062064fab1

The commit message explains: 

>  Remove unused code. This code stopped being used in fc50251f67bcf6b152a75bbdf5b6be3a6f0a5cd4